status/discover.go

```diff
@@ -0,0 +1,62 @@
+package status
+
+import (
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"io/ioutil"
+)
+
+// The Kiali ServiceAccount token.
+var saToken string
+
+var clientFactory kubernetes.ClientFactory
+
+func DiscoverJaeger() (url string, err error) {
+	if config.Get().ExternalServices.Jaeger.URL != "" {
+		return config.Get().ExternalServices.Jaeger.URL, nil
+	}
+	return discoverUrlService(config.Get().ExternalServices.Jaeger.ServiceNamespace, config.Get().ExternalServices.Jaeger.Service)
+}
+
+func DiscoverGrafana() (url string, err error) {
+	if config.Get().ExternalServices.Grafana.URL != "" {
+		return config.Get().ExternalServices.Grafana.URL, nil
+	}
+	return discoverUrlService(config.Get().ExternalServices.Grafana.ServiceNamespace, config.Get().ExternalServices.Grafana.Service)
+}
+
+func discoverUrlService(ns string, service string) (url string, err error) {
```

Comment:
I guess this function should return immediately when we're not on openshift, no? Shouldn't even try to get route with openshift api.

Reply:
you're right It's a draft only tested in openshift I was checking in kubrenetes